src/stash/src/initialize.rs

```diff
@@ -0,0 +1,626 @@
+                    IdAllocValue {
+                        next_id: max(&[
+                            MZ_CATALOG_SCHEMA_ID,
+                            PG_CATALOG_SCHEMA_ID,
+                            PUBLIC_SCHEMA_ID,
+                            MZ_INTERNAL_SCHEMA_ID,
+                            INFORMATION_SCHEMA_ID,
+                        ])
+                        .expect("known to be non-empty")
+                            + 1,
+                    },
+                ),
+                (
+                    IdAllocKey {
+                        name: USER_ROLE_ID_ALLOC_KEY.to_string(),
+                    },
+                    IdAllocValue { next_id: 1 },
+                ),
+                (
+                    IdAllocKey {
+                        name: USER_CLUSTER_ID_ALLOC_KEY.to_string(),
+                    },
+                    IdAllocValue {
+                        next_id: DEFAULT_USER_CLUSTER_ID.inner_id().unwrap() + 1,
+                    },
+                ),
+                (
+                    IdAllocKey {
+                        name: SYSTEM_CLUSTER_ID_ALLOC_KEY.to_string(),
+                    },
+                    IdAllocValue { next_id: 1 },
+                ),
+                (
+                    IdAllocKey {
+                        name: REPLICA_ID_ALLOC_KEY.to_string(),
+                    },
+                    IdAllocValue {
+                        next_id: DEFAULT_REPLICA_ID + 1,
+                    },
+                ),
+                (
+                    IdAllocKey {
+                        name: AUDIT_LOG_ID_ALLOC_KEY.to_string(),
+                    },
+                    IdAllocValue { next_id: 1 },
+                ),
+                (
+                    IdAllocKey {
+                        name: STORAGE_USAGE_ID_ALLOC_KEY.to_string(),
+                    },
+                    IdAllocValue { next_id: 1 },
+                ),
+            ];
+
+            // Wrap each entry with a MigrationAction::Insert.
+            inserts
+                .into_iter()
+                .map(|(key, val)| MigrationAction::Insert(key, val))
+                .collect()
+        })
+        .await?;
+
+    // Collect audit events so we can commit them once at the very end.
+    let mut audit_events = vec![];
+
+    // If provided, generate a new Id and permissions for the bootstrap role.
+    //
+    // Note: Make sure we do this _after_ initializing the ID_ALLOCATOR_COLLECTION.
+    let bootstrap_role = if let Some(role) = &options.bootstrap_role {
+        let user_id = get_and_increment_id(tx, USER_ID_ALLOC_KEY.to_string()).await?;
+        let role_id = RoleId::user(user_id);
+        let role_val = RoleValue {
+            name: role.to_string(),
+            attributes: Some(RoleAttributes {
+                inherit: false,
+                create_db: true,
+                create_cluster: true,
+                create_role: false,
+            }),
```

Comment:
I think it would be better to use
```
RoleAttributes::new()
  .with_create_db()
  .with_create_cluster(),
```

that way we get the correct default values for everything else. For example, `inherit: false` is not valid.

Reply:
Oh, I see that these are the `RoleAttributes` from `objects.rs` and not the original one from the catalog. In that case, `inherit` should be true.

Reply:
I moved the Stash initialization up a layer into the `adapter` crate. The reason being, _what_ is initialized in the Stash isn't necessarily the business of the Stash itself, but the caller. Also, it allowed me to use the existing helpers like `rbac::owner_privilege` which helped de-dupe code and avoid differences like this